src/main/java/com/aws/greengrass/clientdevices/auth/api/Result.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.api;
+
+
+public final class Result<T> {
```

Comment:
I don't know if java have a construct like this but the benefit of having this value object is that now we can better represent invalid errors states and decide to react to them later. A few advantages of using it are:

1) No more scenarios likeDo I throw an error and let the client figure out how to handle it? Do I return null?
2) Failures can be represented in terms of the domain